src/views/AddItem.jsx

```diff
@@ -25,8 +25,18 @@ export function AddItem({ listId }) {
 	const onFormSubmit = (e) => {
 		e.preventDefault();
 		try {
-			addItem(listId, itemData);
-			setSubmitMessage('Item successfully added to your list!');
+			const cleanItemName = itemData.itemName
+				.replace(/[\W_]/g, '')
```

Comment:
`[` and `]` are the start and end of a character set.
`\W` means "non-word", as opposed to `\w` which will match a word.
`_` is the "_" character.
`/ `mark the beginning and end of a regular expression.
`g` means it's a global search.

Found this useful explanation for what is happening on this line as I'm not too familiar with regex. This seems sufficient for filtering the input to get an accurate match.

Reply:
one more note - `\W` expands out (implicitly) to `[^A-Za-z0-9_]`
Regex is efficient but often not as easy to understand - probably a reasonable tradeoff for this use case

Nice place to potentially put in some debouncing